fiboa_cli/convert_utils.py

```diff
@@ -472,10 +473,18 @@ def convert(
                 else:
                     log(f"Column '{key}' not found in dataset, skipping migration", "warning")
 
-        if self.area_factor != 1 and "area" in columns.values():
-            area_key = next(k for k, v in columns.items() if v == "area")
+        area_key = next((k for k, v in columns.items() if v == "area"), None)
+        if self.area_factor != 1 and area_key:
             gdf[area_key] *= self.area_factor
 
+        if self.area_fill_zero:
+            if area_key in gdf.columns:
+                gdf[area_key] = np.where(
+                    gdf[area_key] == 0, gdf.area * self.FACTOR_M2_TO_HA, gdf[area_key]
+                )
+            else:
+                gdf[area_key] = gdf.area * self.FACTOR_M2_TO_HA
+
```

Comment:
gdf,area may be invalid for a geographic CRS using degrees as units; we'd need to use [GeoSeries.to_crs()](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.to_crs.html#geopandas.GeoSeries.to_crs) to project geometries to a planar CRS before using this function. I think we do this already somewhere else in the code, potentially in the improve command, maybe something to centralize.

Reply:
That's fine, but maybe we should error in case the flag is set and doesn't make sense...

Reply:
Hmmm, I would have to check if we can get the SRID-definition. It's in the `proj` library for certain. Will check it out.

Reply:
See https://github.com/fiboa/cli/blob/main/fiboa_cli/improve.py#L85

Reply:
Easier than expected: https://github.com/fiboa/cli/pull/173/commits/06f0d895ba476b20ff636c4e0d9f6cd8263f07c9